app/services/protocol_importers/import_protocol_service.rb

```diff
@@ -0,0 +1,49 @@
+# frozen_string_literal: true
+
+module ProtocolImporters
+  class ImportProtocolService
+    extend Service
+
+    attr_reader :errors, :protocol
+
+    def initialize(protocol_params:, steps_params:, team_id:, user_id:)
+      @user = User.find user_id
+      @team = Team.find team_id
+      @protocol_params = protocol_params
+      @steps_params = steps_params
+      @errors = {}
+    end
+
+    def call
+      return self unless valid?
+
+      @protocol = Protocol.new(@protocol_params.merge!(added_by: @user, team: @team))
+
+      @protocol.steps << @steps_params.collect do |step_params|
+        Step.new(step_params.merge(user: @user, completed: false))
+      end
+
+      errors[:protocol] = @protocol.errors.messages unless @protocol.save
```

Comment:
Should here be `@errors`?

Reply:
Nice 1. You just figure out that also `errors` working because we have set `attr_reader`. But I agree, should use `@` prefix to be consistent